scripts/py_blob.tmpl

```diff
@@ -1,44 +1,49 @@
 """
  Flash OS Routines (Automagically Generated)
- Copyright (c) 2009-2015 ARM Limited
- 
+ Copyright (c) 2017-2017 ARM Limited
+
  Licensed under the Apache License, Version 2.0 (the "License");
  you may not use this file except in compliance with the License.
  You may obtain a copy of the License at
- 
+
      http://www.apache.org/licenses/LICENSE-2.0
- 
+
  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the License.
-""" 
-
-from flash_kinetis import Flash_Kinetis
-
-flash_algo = { 
-    'load_address' : {{entry}},
-    'instructions' : [
-    {{prog_header}}
-    {{mem}}
-    ],
-    
-    'pc_init' : {{func['Init']}},
-    'pc_unInit': {{func['UnInit']}},
-    'pc_program_page': {{func['ProgramPage']}},
-    'pc_erase_sector': {{func['EraseSector']}},
-    'pc_eraseAll' : {{func['EraseChip']}},
-
-    'static_base' : {{entry}} + {{header_size}} + {{static_base}},
-    'begin_stack' : {{entry}} + 0x00000800,            
-    'begin_data' : {{entry}} + 0x00000A00,
-    'page_size' : {{prog_page_size}},
-    'analyzer_supported' : False,
-    'analyzer_address' : 0x00000000  # ITCM, Analyzer 0x00000000..0x000000600
-};
-              
-class Flash_{{name}}(Flash_Kinetis): 
-    def __init__(self, target):
-        super(Flash_{{name}}, self).__init__(target, flash_algo)
+"""
+
+flash_algo = {
+
+    # Flash algorithm as a hex string
+    'instructions':
+        {{algo.format_algo_data(8, 64, "hex")}},
+
+    # Relative function addresses
+    'pc_init': {{'0x%x' % algo.symbols['Init']}},
+    'pc_unInit': {{'0x%x' % algo.symbols['UnInit']}},
+    'pc_program_page': {{'0x%x' % algo.symbols['ProgramPage']}},
+    'pc_erase_sector': {{'0x%x' % algo.symbols['EraseSector']}},
+    'pc_eraseAll': {{'0x%x' % algo.symbols['EraseChip']}},
+
+    # Relative region addresses and sizes
+    'ro_start': {{'0x%x' % algo.ro_start}},
+    'ro_size': {{'0x%x' % algo.ro_size}},
+    'rw_start': {{'0x%x' % algo.rw_start}},
+    'rw_size': {{'0x%x' % algo.rw_size}},
+    'zi_start': {{'0x%x' % algo.zi_start}},
+    'zi_size': {{'0x%x' % algo.zi_size}},
+
+    # Flash information
+    'flash_start': {{'0x%x' % algo.flash_start}},
+    'flash_size': {{'0x%x' % algo.flash_size}},
+    'page_size': {{'0x%x' % algo.page_size}},
+    'sector_sizes': (
+    {%- for start, size  in algo.sector_sizes %}
+        {{ "(0x%x, 0x%x)" % (start, size) }}, 
+    {%- endfor %}
+    )
+}
```

Comment:
Actually, I have a [pyOCD branch](https://github.com/c1728p9/pyOCD/tree/algo_gen) that  computes the analyzer and buffer addresses dynamically [here](https://github.com/c1728p9/pyOCD/blob/algo_gen/pyOCD/flash/flash.py#L174).  That way the layout in memory can change without needing to regenerate the algos. Would this instead accomplish your goal?